rfd/0087-access-request-notification-routing.md

```diff
@@ -0,0 +1,207 @@
+In such situation, `access_routing_rules` should be considered as critical as `roles` themselves as they can lead
+to automatic access escalation. Such risk is already present in the current implementation.
+
+This risk can be mitigated by not granting the access-request edition rights to the plugin (at the price of a feature loss).
+
+If the two targets are identical it is the plugin's responsibility to deduplicate before sending notifications
+(to avoid sending duplicates and to ensure `PluginData` consistence).
+
+### User Experience
+
+#### Complexity/Time to value
+
+In most cases, a simple role_name check is performed and the users wants to set the plugin and the recipient.
+This is provided by the simplified syntax:
+
+```yaml
+targets:
+  - condition: 'equals(resource.spec.requested_role, "prod-rw")'
+    recipient: "Teleport Alice"
+    plugin: "pagerduty"
+```
+
+This syntax is equivalent to the pure-predicate one:
+
+```yaml
+taregts:
+  - expression: >
+      ifelse(
+        equals(resource.spec.requested_role, "prod-rw"),
+        pair("pagerduty", "Teleport Alice"),
+        pair()
+      )
+```
+
+If users want to do complex operations and leverage all the predicate language power they can use pass the predicate code
+in the `expression` field. The predicate evaluation in `expression` field is expected to return a pair containing the plugin
+name and the recipient.b If one or both fields are empty, the target should not be set.
+
+If `expression` is set, `condition`, `recipient` and `plugin` should not be used. Teleport must validate
+this on resource creation and fail immediately instead of silently ignoring fields.
+
+#### Combining with suggested_reviewers
+
+The safest way would be for this new feature to not interact with suggested reviewers and let the plugins deal with it.
+
+Using `suggested_reviwer` implies the user knows which plugin will be used to send the access request. 
+The string has to make sense and be allowed by the plugin. Today most plugins support email addresses
+and reject other recipients, but somes are just ignoring the field (pagerduty). We might want to clarify
+what can be entered in this field and how different plugins are reacting to it so user has insights on
+what to type and what to expect.
+
+Using `suggested_reviewer` with two plugins supporting it (e.g. Slack and MsTeams, or two Slack plugins on different workspaces)
+can lead to the suggested reviewer being notified multiple times. One way to mitigate this would be to
+offer a plugin configuration flag to ignore suggested reviewers (e.g. suggested reviewers are only
+pinged on Slack).
+
+#### JIT resource access request
+
+While `access_rouring_rule` should work with both role access requests and resource access requests,
+using it withresource access request can cause some friction for the admins writing the rules:
+
+Only resource IDs are available through access requests: writing a rule to filter based on requested resources
```

Comment:
FYI @nklaassen is adding some additional info to access requests to allow us to show actual resource names instead of IDs - Nic, could you link an RFD/PR (if there's any) here so we can see if it will work with the plugin routing rules?

Reply:
@nklaassen I think the label-based routing is the best answer on how to support resource based access requests with this syntax.